mar/graph/validator/library.hoon

```diff
@@ -0,0 +1,101 @@
+/-  *post, met=metadata-store, graph=graph-store
+=>
+|%
+++  is-title-valid
+  |=  [title=tape]
+  ^-  ?
+  %.y
+++  is-isbn-valid  :: we could do https://github.com/xlcnd/isbnlib/blob/41f59c74a69a2675c3f135431e9785f9ae502a7e/isbnlib/_core.py#L52
+  |=  [isbn=tape]
+  ^-  ?
+  =/  len  (lent isbn)
+  ?>  ?|(=(len 10) =(len 13))
+  %.y
+--
+|_  i=indexed-post
+++  grow
+  |%
+  ++  noun  i
+  ::
+  ++  graph-indexed-post
+    ^-  indexed-post
+    ?+    index.p.i  !!
+    ::  top level node: book
+    ::  structural node with no content
+    ::
+        [@ ~]
+      ~|  "top level book node should be empty!"
+      ?>  ?=(~ contents.p.i)
+      i
+    ::  metadata revision container
+    ::  structural node with no content
+    ::
+        [@ %meta ~]
+      ~|  "metadata revision container should be empty!"
+      ?>  ?=(~ contents.p.i)
+      i
+    ::  single metadata revision
+    ::  content node. first %text element is treated as title,
+    ::  second is treated as isbn
+    ::
+        [@ %meta @ ~]
+      =/  contents  contents.p.i
+      ?>  ?=([[%text *] [%text *] ~] contents.p.i)
+      =/  title  (trip +.i.contents.p.i)
+      ?>  (is-title-valid title)
+      =/  isbn  (trip +.i.t.contents.p.i)
+      ?>  (is-isbn-valid isbn)
+      i
+    ::  comments section container
+    ::  structural node with no content
+    ::
+        [@ %comments ~]
+      ~|  "comments section container should be empty!"
+      ?>  ?=(~ contents.p.i)
+      i
+    ::  comment
+    ::
+        [@ %comments @ ~]
+      :: we could allow any content eventually
+      ::?>  ?=(* contents.p.i)  :: any content is allowed.
+      ?>  ?=([[%text *] ~] contents.p.i)  :: only a single %text content is allowed
+      =/  contents  contents.p.i
+      i
+    ==
+  ::  vestigial structures from graph-push-hook. don't know if we still need these
+  ++  graph-permissions-add
+    |=  vip=vip-metadata:met
+    ^-  permissions:graph
+    !!
+  ::
+  ++  graph-permissions-remove
+    |=  vip=vip-metadata:met
+    ^-  permissions:graph
+    !!
+  ++  transform-add-nodes
```

Comment:
If you aren't using graph-push-hook, you don't need this. (Unless you are using it in your proxy-hook!)

Reply:
Well, as of now I'm not using it, but it does bring up the question of if we *should* be using it. IIRC this arm was basically an extra step that would ensure all time based indexes of an incoming post were modified to be when they were received on the host, as opposed to trusting the author. If this API is here to stay, and it's a best practice in your eyes, then I can try to figure out which places it would make sense to run an update through the arm.

Reply:
OOB: Logan: complexity of transform add nodes is not justified for this example, we don't need its functionality anyways, so its good to remove

Reply:
https://github.com/ynx0/library/pull/3/commits/bccc429ef0d6c182f720201d5c5a7a8d31610019